### PR TITLE
Fixes hashivault connection check

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
* Initially in the https://github.com/tektoncd/chains/pull/933 it was checking if the vault address is valid or not, but if the url did not have port then it used to return an error and could not proceed further for signing

* Hence this patch fixes this by checking if the url has port or if the url is a http or https url and then validates if the url is valid so that it can proceed for signing

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

``` release-note
NONE
```

### CODE REVIEW

File: pkg/chains/signing/kms/kms.go [hunk 1]
```diff
@@ -51,11 +52,35 @@ func NewSigner(ctx context.Context, cfg config.KMSSigner) (*Signer, error) {
...
+				Host:   vaultAddress.Host + ":443",
+			}
+		case vaultAddress.Scheme == "":
+			vaultUrl = &url.URL{
+				Scheme: "http",
+				Host:   cfg.Auth.Address + ":80",
+			}
+		}
+
+		if vaultUrl != nil {
```
Comment (lcarva) [inline pkg/chains/signing/kms/kms.go:76]: Shouldn't we return an error if it is `nil`?

Comment (PuneetPunamiya) [reply pkg/chains/signing/kms/kms.go:76]: as the error will be nil right ?

Comment (lcarva) [reply pkg/chains/signing/kms/kms.go:76]: Right. You want to create a new error in this case.

File: pkg/chains/signing/kms/kms.go [hunk 2]
```diff
@@ -51,11 +52,35 @@ func NewSigner(ctx context.Context, cfg config.KMSSigner) (*Signer, error) {
 			return nil, err
 		}
 
-		conn, err := net.DialTimeout("tcp", vaultAddress.Host, 5*time.Second)
-		if err != nil {
-			return nil, err
+		var vaultUrl *url.URL
+		switch {
```
[the remark this answers is not in the document]
Comment (vdemeester) [reply pkg/chains/signing/kms/kms.go:56]: Can't we rely on a default scheme ? (`https://` or something ?)

**Review by vdemeester — APPROVED** [review]

File: pkg/chains/signing/kms/kms.go [hunk 3]
```diff
@@ -51,11 +52,45 @@ func NewSigner(ctx context.Context, cfg config.KMSSigner) (*Signer, error) {
...
+			}
+			if vaultUrl.Port() == "" {
+				vaultUrl.Host = cfg.Auth.Address + ":80"
+			}
+		}
+
+		if vaultUrl != nil {
+			conn, err := net.DialTimeout("tcp", vaultUrl.Host, 5*time.Second)
+			if err != nil {
+				fmt.Printf("Error connecting to URL %s: %v\n", vaultUrl, err)
```
Comment (lcarva) [inline pkg/chains/signing/kms/kms.go:87]: Let's not print here.

File: pkg/chains/signing/kms/kms.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (PuneetPunamiya) [reply pkg/chains/signing/kms/kms.go:56]: Won't this case cover this scenario https://github.com/tektoncd/chains/pull/936/files#diff-63cec1a1260227d38ef0e60ce232822a3f65f580869912386619e2c8d66ce191R74-R82 🤔

[the remark this answers is not in the document]
Comment (lcarva) [reply pkg/chains/signing/kms/kms.go:56]: Yup. Missed that.

**Pull request merged**